From 921c8f1953bedba9052bc686af46dd199f0ce57c Mon Sep 17 00:00:00 2001
From: Aakash Sasidharan <asasidharan@marvell.com>
Date: Fri, 16 Feb 2024 16:44:59 +0530
Subject: [PATCH 273/513] test/crypto: add combined mode cases for TLS 1.3

Add cases to try TLS 1.3 record write(encrypt) + read(decrypt)
operations. This is used for testing TLS 1.3 record features with
all algorithms supported by the security device.

Signed-off-by: Aakash Sasidharan <asasidharan@marvell.com>
Change-Id: I14cce2b4eca7895842c5698bd502e23aedf18cf0
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/dataplane/dpdk/+/122088
Base-Builds: sa_ip-toolkits-Jenkins <sa_ip-toolkits-jenkins@marvell.com>
Tested-by: sa_ip-toolkits-Jenkins <sa_ip-toolkits-jenkins@marvell.com>
Reviewed-by: Anoob Joseph <anoobj@marvell.com>
---
 app/test/test_cryptodev.c | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/app/test/test_cryptodev.c b/app/test/test_cryptodev.c
index 4a02fc2d19700..7ec4dadb5e3bc 100644
--- a/app/test/test_cryptodev.c
+++ b/app/test/test_cryptodev.c
@@ -12611,6 +12611,19 @@ test_dtls_1_2_record_proto_sg_opt_padding_max(void)
 	return test_tls_record_proto_opt_padding(33, 4, RTE_SECURITY_VERSION_DTLS_1_2);
 }
 
+static int
+test_tls_1_3_record_proto_display_list(void)
+{
+	struct tls_record_test_flags flags;
+
+	memset(&flags, 0, sizeof(flags));
+
+	flags.display_alg = true;
+	flags.tls_version = RTE_SECURITY_VERSION_TLS_1_3;
+
+	return test_tls_record_proto_all(&flags);
+}
+
 static int
 test_tls_1_3_record_proto_corrupt_pkt(void)
 {
@@ -18040,6 +18053,10 @@ static struct unit_test_suite tls13_record_proto_testsuite  = {
 			"TLS-1.3 record SG mode with optional padding",
 			ut_setup_security, ut_teardown,
 			test_tls_1_3_record_proto_sg_opt_padding_1),
+		TEST_CASE_NAMED_ST(
+			"Combined test alg list",
+			ut_setup_security, ut_teardown,
+			test_tls_1_3_record_proto_display_list),
 		TEST_CASES_END() /**< NULL terminate unit test array */
 	}
 };
-- 
2.25.1

